compiler/bindings/c/iree/compiler/dialects/iree_gpu.h

```diff
@@ -144,6 +144,18 @@ struct ireeGPUMMASingleSubgroupLayout {
 MLIR_CAPI_EXPORTED ireeGPUMMASingleSubgroupLayout
 ireeGPUGetSingleSubgroupLayout(MlirAttribute attr, uint32_t fragment);
 
+struct ireeGPUTargetInfo {
+  MlirIdentifier arch;                 // "gfx942".
+  MlirAttribute subgroup_size_choices; // Subgroup size choices.
+  MlirAttribute max_workgroup_sizes;   // Max threads per X/Y/Z dimension.
+  MlirAttribute max_thread_count_per_workgroup; // Max threads per workgroup.
+  MlirAttribute max_workgroup_memory_bytes;     // Max workgroup memory.
+};
+
+// Add function to query GPU target info from ExecutableTargetAttr.
```

Comment:
This doesn't add any functions, this is a function to query something
```suggestion
// Queries GPU target info from the given `ExecutableTargetAttr` attribute |attr|.
```